### PR TITLE
Wagtail 2.16 upgrade

### LINKED ISSUE #472: Upgrade to Wagtail 2.16
This might be helpful: https://wagtail.org/blog/wagtail-216-and-upcoming-breaking-changes/

### PR DESCRIPTION
**Pull request by rcoldiron**
Fixes #472

### CODE REVIEW

File: docs/releases/v0.23.0.rst [hunk 1]
```diff
@@ -19,13 +19,23 @@ Maintenance
...
+* Wagtail dropped support for Python 3.6. Upgrade your Python to 3.7 and above.
+
+* Wagtail dropped support for Django 3.0, 3.1, and 3.2. Upgrade your Django to 3.2 and above.
+
+* Wagtail is now using a new search backend.
+
+    * Specify the `new default in your settings <https://docs.wagtail.org/en/stable/releases/2.15.html#database-search-backends-replaced>`_.
+    * Next, you will need to run the `update_index command <https://docs.wagtail.org/en/stable/reference/management_commands.html#update-index>`_.
+
+* Other upgrade considerations for `Wagtail 2.15 <https://docs.wagtail.org/en/stable/releases/2.15.html#upgrade-considerations>`_ and `Wagtail 2.16 <https://docs.wagtail.org/en/stable/releases/2.16.html#upgrade-considerations>`_ may apply to your site.
```
Comment (vsalvino) [inline docs/releases/v0.23.0.rst:31]: Did you run into any migration issues, i.e. does `makemigrations` product any output or try to make anything?

File: docs/releases/v0.23.0.rst [hunk 2]
```diff
@@ -19,13 +19,23 @@ Maintenance
 Upgrade considerations
 ----------------------
 
+* Wagtail dropped support for Python 3.6. Upgrade your Python to 3.7 and above.
+
+* Wagtail dropped support for Django 3.0, 3.1, and 3.2. Upgrade your Django to 3.2 and above.
```
Comment (vsalvino) [inline docs/releases/v0.23.0.rst:24]: Typo about dropping 3.2 and upgrade to 3.2

**Comment by vsalvino** [comment]
In the unit test output, I'm seeing a lot of Removed in Django 4.0 warnings. Can we clean some of those up? I'm not sure if our unit tests are using 4.0 or 3.2, but it might be best to run them with both, or at least with the newer version.

File: docs/releases/v0.23.0.rst (hunk 1 above)
Comment (rcoldiron) [reply docs/releases/v0.23.0.rst:31]: No migrations for codered cms. All the migrations ran in my local wagtail package.

**Comment by rcoldiron** [comment]
NOTE: As of 2/16/2022, Wagtail 2.15 and 2.16 new search backend and update_index command work properly with sqlite and postgres when a user puts an emoji in the content but causes an error when using mysql. Changing the encoding and collation does not fix it.

**Comment by rcoldiron** [comment]
STEPS FOR MYSQL ERROR:
1. After creating a new Codered CMS site using this branch's version, set up the MySQL database and connect it, run migrations then the update_index command. 
2. Create your new page models with text fields, make migrations, migrate. Create superuser as normal.
3. Now run `manage.py update_index`.
4. Run the server and add a page. In a text field, add an emoji character and hit publish. The error in local environment is: (1366, "Incorrect string value: '\\xF0\\x9F\\x98\\x81' for column `testsite`.`wagtailsearch_indexentry`.`body` at row 1")
5. Trying to run update_index again will also show this error.

NOTE: This does not occur in standard Wagtail 2.15.* new projects with MySQL. It must be the way the new search backend or the update_index command interacts with something in Codered CMS page types or fields or search functions.

**Pull request merged**